cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -43,9 +43,9 @@ type MasterConfiguration struct {
 }
 
 type API struct {
-	AdvertiseAddresses []string
-	ExternalDNSNames   []string
-	Port               int32
+	BindAddress      string
```

Comment:
Actually -- I don't know if this works if we don't listen on localhost as the admission controllers talk to the API server on a loopback address.  If we turn off the insecure port then it *has* to talk to the secure port.  Worth testing.